src/socket/unix_server.cr

```diff
@@ -1,82 +1,163 @@
 require "./unix_socket"
+require "./server"
 
-# A local interprocess communication server socket.
+# A local interprocess communication (UNIX socket) server socket.
 #
 # Only available on UNIX and UNIX-like operating systems.
 #
-# Example usage:
+# Usage example:
 # ```
-# require "socket"
+# require "socket/unix_server"
 #
 # def handle_client(client)
 #   message = client.gets
 #   client.puts message
 # end
 #
-# server = UNIXServer.new("/tmp/myapp.sock")
-# while client = server.accept?
-#   spawn handle_client(client)
+# UNIXServer.open("/tmp/myapp.sock") do |server|
+#   while client = server.accept?
+#     spawn handle_client(client)
+#   end
 # end
 # ```
-class UNIXServer < UNIXSocket
+struct UNIXServer
   include Socket::Server
 
-  # Creates a named UNIX socket, listening on a filesystem pathname.
+  # Returns the raw socket wrapped by this UNIX server.
+  getter raw : Socket::Raw
+
+  @address : Socket::UNIXAddress
+
+  # Creates a `UNIXServer` from a raw socket.
+  def initialize(@raw : Socket::Raw, @address : Socket::UNIXAddress)
+  end
+
+  # Creates a named UNIX socket listening on a filesystem pathname.
   #
   # Always deletes any existing filesystam pathname first, in order to cleanup
   # any leftover socket file.
   #
-  # The server is of stream type by default, but this can be changed for
-  # another type. For example datagram messages:
   # ```
-  # UNIXServer.new("/tmp/dgram.sock", Socket::Type::DGRAM)
```

Comment:
how is a `UNIXServer` created with `DRAM` type? Some docs left use `dgram.sock` but I am a bit lost how is that functionality preserved.

Reply:
Using `UNIXServer`/`UNIXSocket` with DGRAM type is no longer supported. It would be too weird and contradict the purpose of this PR if a stream socket (implementing `IO`) could be used as a datagram socket.

We could add an additional type for this but I don't think this use case is anywhere common enough for that. If you want to send datagrams over UNIX sockets, you should simply use `Socket::Raw`.